rticonnextdds_connector_test.go

```diff
@@ -160,64 +161,137 @@ func TestDataFlow(t *testing.T) {
 
-	assert.Equal(t, input.Samples.GetString(0, "st"), st)
-	assert.Equal(t, input.Samples.GetBoolean(0, "b"), b)
-
-	assert.Equal(t, input.Samples.GetUint8(0, "c"), c)
-	assert.Equal(t, input.Samples.GetByte(0, "c"), c)
-	assert.Equal(t, input.Samples.GetInt8(0, "xs"), xs)
-	assert.Equal(t, input.Samples.GetInt16(0, "s"), s)
-	assert.Equal(t, input.Samples.GetUint16(0, "us"), us)
-	assert.Equal(t, input.Samples.GetInt32(0, "l"), l)
-	assert.Equal(t, input.Samples.GetInt(0, "l"), int(l))
-	assert.Equal(t, input.Samples.GetInt64(0, "ll"), ll)
-	assert.Equal(t, input.Samples.GetUint(0, "ul"), uint(ul))
-	assert.Equal(t, input.Samples.GetRune(0, "l"), rune(l)) //nolint //this is because "l" rune is still an int32
-	assert.Equal(t, input.Samples.GetUint32(0, "ul"), ul)
-	assert.Equal(t, input.Samples.GetUint64(0, "ull"), ull)
-	assert.Equal(t, input.Samples.GetFloat32(0, "f"), f)
-	assert.Equal(t, input.Samples.GetFloat64(0, "d"), d)
+	rst, err := input.Samples.GetString(0, "st")
+	assert.Nil(t, err)
+	assert.Equal(t, rst, st)
+
+	rb, err := input.Samples.GetBoolean(0, "b")
+	assert.Nil(t, err)
+	assert.Equal(t, rb, b)
+
+	rc, err := input.Samples.GetByte(0, "c")
+	assert.Nil(t, err)
+	assert.Equal(t, rc, c)
+
+	rxs, err := input.Samples.GetInt8(0, "xs")
+	assert.Nil(t, err)
+	assert.Equal(t, rxs, xs)
+
+	rc, err = input.Samples.GetUint8(0, "c")
+	assert.Nil(t, err)
+	assert.Equal(t, rc, c)
+
+	rs, err := input.Samples.GetInt16(0, "s")
+	assert.Nil(t, err)
+	assert.Equal(t, rs, s)
+
+	rus, err := input.Samples.GetUint16(0, "us")
+	assert.Nil(t, err)
+	assert.Equal(t, rus, us)
+
+	rl, err := input.Samples.GetInt32(0, "l")
+	assert.Nil(t, err)
+	assert.Equal(t, rl, l)
+
+	rul, err := input.Samples.GetUint32(0, "ul")
+	assert.Nil(t, err)
+	assert.Equal(t, rul, ul)
+
+	ri, err := input.Samples.GetInt(0, "l")
+	assert.Nil(t, err)
+	assert.Equal(t, ri, int(l))
+
+	rui, err := input.Samples.GetUint(0, "ul")
+	assert.Nil(t, err)
+	assert.Equal(t, rui, uint(ul))
+
+	rl, err = input.Samples.GetRune(0, "l")
+	assert.Nil(t, err)
+	assert.Equal(t, rl, l)
+
+	rll, err := input.Samples.GetInt64(0, "ll")
+	assert.Nil(t, err)
+	assert.Equal(t, rll, ll)
+
+	rull, err := input.Samples.GetUint64(0, "ull")
+	assert.Nil(t, err)
+	assert.Equal(t, rull, ull)
+
+	rf, err := input.Samples.GetFloat32(0, "f")
+	assert.Nil(t, err)
+	assert.Equal(t, rf, f)
+
+	rd, err := input.Samples.GetFloat64(0, "d")
+	assert.Nil(t, err)
+	assert.Equal(t, rd, d)
 
 	assert.Nil(t, output.ClearMembers())
 
 	// Testing Wait TimeOut
 	assert.NotNil(t, connector.Wait(5))
 
 	// Testing Read
-	assert.Nil(t, output.Write())
-	assert.Nil(t, connector.Wait(-1))
-	assert.Nil(t, input.Read())
-
-	assert.NotEqual(t, input.Samples.GetString(0, "st"), st)
+	err = output.Write()
+	assert.Nil(t, err)
+	err = connector.Wait(-1)
+	assert.Nil(t, err)
+	err = input.Read()
+	assert.Nil(t, err)
+	rst, err = input.Samples.GetString(0, "st")
+	assert.Nil(t, err)
+	assert.Equal(t, rst, "")
```

Comment:
Wait here too, can you fix it? inline assert should be fine here.

Reply:
Can we do a single line assert for a function having multiple return values?

Reply:
oh wait sorry my bad. for some reason i see it differently..